metals/src/main/scala/scala/meta/internal/metals/Compilers.scala

```diff
@@ -226,7 +226,7 @@ class Compilers(
       ds.asScala.headOption match {
         case None =>
           diagnostics.onNoSyntaxError(path)
-        case Some(diagnostic) =>
+        case Some(diagnostic) if !path.isInReadonlyDirectory(workspace) =>
```

Comment:
I also added a small change to not publish syntax errors in library sources. It seems to happen sporadically for Scala 3 sources.